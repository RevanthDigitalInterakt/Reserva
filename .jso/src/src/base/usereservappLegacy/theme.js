  Object.defineProperty(exports, "__esModule", {
    value: true
  });
  exports.theme = undefined;
  var tokens = _interopRequireWildcard(_$$_REQUIRE(_dependencyMap[0]));
  function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(e) { return e ? t : r; })(e); }
  function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
  var theme = exports.theme = {
    space: {
      quarck: 4,
      nano: 8,
      micro: 12,
      xxxs: 16,
      xxs: 24,
      xs: 32,
      sm: 40,
      md: 48,
      xl: 64,
      xxl: 80
    },
    shadows: {
      bottomBarShadow: '0px -3px 2px rgba(0,0,0,0.1)',
      topBarShadow: '0px 3px 2px rgba(0,0,0,0.1)'
    },
    colors: {
      preto: tokens.preto,
      vermelhoFechadoRSV: tokens.vermelhoFechadoRSV,
      marinho: tokens.marinho,
      neutroFrio2: tokens.neutroFrio2,
      neutroQuente2: tokens.neutroQuente2,
      vermelhoRSV: tokens.vermelhoRSV,
      verde: tokens.verde,
      neutroQuente1: tokens.neutroQuente1,
      offWhite: tokens.offWhite,
      offWhite15: tokens.offWhite15,
      whiteSecondary: tokens.whiteSecondary,
      neutroFrio1: tokens.neutroFrio1,
      leve: tokens.leve,
      vermelhoAlerta: tokens.vermelhoAlerta,
      verdeSucesso: tokens.verdeSucesso,
      amareloAtencao: tokens.amareloAtencao,
      dropDownBorderColor: tokens.dropDownBorderColor,
      white: tokens.white,
      fullBlack: tokens.fullBlack,
      divider: tokens.divider,
      backgoundInput: tokens.backgoundInput,
      backgroundApp: tokens.backgroundApp,
      transparente: 'transparent',
      modalBackDropColor: tokens.modalBackDropColor,
      borderButton: tokens.borderButton,
      dividerCounter: tokens.dividerCounter,
      backgoundDivider: tokens.backgoundDivider,
      progressTextColor: tokens.progressTextColor,
      backgroundMenuOpened: tokens.backgroundMenuOpened,
      searchBarTextColor: tokens.searchBarTextColor,
      borderColorProduct: tokens.borderColorProduct
    },
    fonts: {
      nunitoBlack: 'Nunito-Black',
      nunitoBlackItalic: 'Nunito-BlackItalic',
      nunitoBold: 'Nunito-Bold',
      nunitoBoldItalic: 'Nunito-BoldItalic',
      nunitoExtraBold: 'Nunito-ExtraBold',
      nunitoExtraBoldItalic: 'Nunito-ExtraBoldItalic',
      nunitoExtraLight: 'Nunito-ExtraLight',
      nunitoExtraLightItalic: 'Nunito-ExtraLightItalic',
      nunitoItalic: 'Nunito-Italic',
      nunitoLight: 'Nunito-Light',
      nunitoLightItalic: 'Nunito-LightItalic',
      nunitoRegular: 'Nunito-Regular',
      nunitoSemiBold: 'Nunito-SemiBold',
      nunitoSemiBoldItalic: 'Nunito-SemiBoldItalic',
      reservaDisplayRegular: 'ReservaDisplay-Regular',
      reservaSerifBlack: 'ReservaSerif-Black',
      reservaSerifBlackItalic: 'ReservaSerif-BlackItalic',
      reservaSerifBold: 'ReservaSerif-Bold',
      reservaSerifBoldItalic: 'ReservaSerif-BoldItalic',
      reservaSerifItalic: 'ReservaSerif-Italic',
      reservaSerifLight: 'ReservaSerif-Light',
      reservaSerifLightItalic: 'ReservaSerif-LightItalic',
      reservaSerifMedium: 'ReservaSerif-Medium',
      reservaSerifMediumItalic: 'ReservaSerif-MediumItalic',
      reservaSerifRegular: 'ReservaSerif-Regular',
      reservaSerifThin: 'ReservaSerif-Thin',
      reservaSerifThinItalic: 'ReservaSerif-ThinItalic',
      reservaSansBlack: 'ReservaSans-Black',
      reservaSansBlackItalic: 'ReservaSans-BlackItalic',
      reservaSansBold: 'ReservaSans-Bold',
      reservaSansBoldItalic: 'ReservaSans-BoldItalic',
      reservaSansItalic: 'ReservaSans-Italic',
      reservaSansLight: 'ReservaSans-Light',
      reservaSansLightItalic: 'ReservaSans-LightItalic',
      reservaSansMedium: 'ReservaSans-Medium',
      reservaSansMediumItalic: 'ReservaSans-MediumItalic',
      reservaSansRegular: 'ReservaSans-Regular',
      reservaSansThin: 'ReservaSans-Thin',
      reservaSansThinItalic: 'ReservaSans-ThinItalic'
    },
    fontWeights: {
      light: 300,
      normal: 400,
      medium: 500,
      SemiBold: 600,
      bold: 700,
      black: 900
    },
    fontSizes: [4, 8, 12, 14, 16, 18, 20, 24, 27, 32, 48],
    radii: {
      pico: 2,
      nano: 5,
      xxxs: 10,
      xs: 15,
      sm: 20,
      pill: 500,
      infinity: 999999
    },
    borderWidths: {
      none: tokens.borderWidthNone,
      hairline: tokens.borderWidthHairline,
      thin: tokens.borderWidthThin,
      thick: tokens.borderWidthThick,
      heavy: tokens.borderWidthHeavy,
      borderDiscountTag: tokens.borderDiscountTag,
      stepper: 3
    }
  };
