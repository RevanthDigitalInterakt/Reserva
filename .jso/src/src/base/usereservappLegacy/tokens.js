  Object.defineProperty(exports, "__esModule", {
    value: true
  });
  exports.whiteSecondary = exports.white = exports.vermelhoRSV = exports.vermelhoFechadoRSV = exports.vermelhoAlerta = exports.verdeSucesso = exports.verde = exports.searchBarTextColor = exports.progressTextColor = exports.preto = exports.offWhite15 = exports.offWhite = exports.neutroQuente2 = exports.neutroQuente1 = exports.neutroFrio2 = exports.neutroFrio1 = exports.modalBackDropColor = exports.marinho = exports.leve = exports.fullBlack = exports.dropDownBorderColor = exports.dividerCounter = exports.divider = exports.borderWidthThin = exports.borderWidthThick = exports.borderWidthNone = exports.borderWidthHeavy = exports.borderWidthHairline = exports.borderDiscountTag = exports.borderColorProduct = exports.borderButton = exports.backgroundMenuOpened = exports.backgroundApp = exports.backgoundInput = exports.backgoundDivider = exports.amareloAtencao = undefined;
  var white = exports.white = '#FFF';
  var fullBlack = exports.fullBlack = '#000';

  // cores primarias
  var preto = exports.preto = '#333333';
  var vermelhoFechadoRSV = exports.vermelhoFechadoRSV = '#5A0000';
  var marinho = exports.marinho = '#13294B';
  var neutroFrio2 = exports.neutroFrio2 = '#8A8C8E';
  var neutroQuente2 = exports.neutroQuente2 = '#B1ACA8';
  var divider = exports.divider = '#D8D9DA';
  var dividerCounter = exports.dividerCounter = '#EAEAEA';
  var progressTextColor = exports.progressTextColor = '#979797';
  var searchBarTextColor = exports.searchBarTextColor = '#666666';

  // cores secundarias
  var vermelhoRSV = exports.vermelhoRSV = '#A00000';
  var verde = exports.verde = '#22372B';
  var neutroQuente1 = exports.neutroQuente1 = '#D7CFCA';
  var offWhite = exports.offWhite = '#D6D1C4';
  var offWhite15 = exports.offWhite15 = '#F9F8F6';
  var whiteSecondary = exports.whiteSecondary = '#FAFAFA';
  var neutroFrio1 = exports.neutroFrio1 = '#D8D9DA';
  var leve = exports.leve = '#F9E1DD';

  // cores do sistema
  var vermelhoAlerta = exports.vermelhoAlerta = '#EF1E1E';
  var verdeSucesso = exports.verdeSucesso = '#38A238';
  var amareloAtencao = exports.amareloAtencao = '#FEA528';
  var dropDownBorderColor = exports.dropDownBorderColor = '#DDDDDD';

  // larguras de borda
  var borderWidthNone = exports.borderWidthNone = 0;
  var borderWidthHairline = exports.borderWidthHairline = 1;
  var borderWidthThin = exports.borderWidthThin = 2;
  var borderWidthThick = exports.borderWidthThick = 4;
  var borderWidthHeavy = exports.borderWidthHeavy = 8;
  var borderDiscountTag = exports.borderDiscountTag = 12;

  // cores de borda
  var borderButton = exports.borderButton = '#333333';
  var borderColorProduct = exports.borderColorProduct = '#757575';

  // backgrounds
  var backgroundMenuOpened = exports.backgroundMenuOpened = '#F1F1F1';
  var backgroundApp = exports.backgroundApp = '#F9F9F9';
  var modalBackDropColor = exports.modalBackDropColor = '#AFAFAF';
  var backgoundInput = exports.backgoundInput = '#F0F0F0';
  var backgoundDivider = exports.backgoundDivider = '#F5F5F5';
