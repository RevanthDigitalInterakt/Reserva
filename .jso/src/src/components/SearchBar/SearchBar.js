  Object.defineProperty(exports, "__esModule", {
    value: true
  });
  exports.SearchBar = SearchBar;
  var _react = _$$_REQUIRE(_dependencyMap[0])(_$$_REQUIRE(_dependencyMap[1]));
  var _reactNative = _$$_REQUIRE(_dependencyMap[2]);
  function SearchBar(_ref) {
    var _ref$height = _ref.height,
      height = _ref$height === undefined ? 36 : _ref$height,
      iconName = _ref.iconName,
      placeholder = _ref.placeholder,
      autocomplete = _ref.autocomplete,
      value = _ref.value,
      onValueChange = _ref.onValueChange,
      onClickIcon = _ref.onClickIcon,
      onClickAutocomplete = _ref.onClickAutocomplete,
      testID = _ref.testID;
    return /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[3]).jsxs)(_$$_REQUIRE(_dependencyMap[4]).Box, {
      children: [/*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[3]).jsxs)(_$$_REQUIRE(_dependencyMap[4]).Box, {
        height: height,
        flexDirection: "row",
        justifyContent: "space-between",
        alignItems: "center",
        borderRadius: "nano",
        backgroundColor: "backgoundInput",
        paddingX: "nano",
        children: [/*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[3]).jsx)(_$$_REQUIRE(_dependencyMap[4]).Box, {
          ml: "quarck",
          mr: "nano",
          children: /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[3]).jsx)(_$$_REQUIRE(_dependencyMap[5]).IconLegacy, {
            name: iconName || 'Search',
            size: 18
          })
        }), /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[3]).jsx)(_$$_REQUIRE(_dependencyMap[4]).Box, {
          flexGrow: 1,
          children: /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[3]).jsx)(_reactNative.TextInput, {
            placeholderTextColor: _$$_REQUIRE(_dependencyMap[6]).theme.colors.searchBarTextColor,
            placeholder: placeholder,
            returnKeyType: "search",
            enablesReturnKeyAutomatically: true,
            onSubmitEditing: function onSubmitEditing() {
              return onClickIcon && onClickIcon();
            },
            onChangeText: function onChangeText(text) {
              if (onValueChange) {
                onValueChange(text);
              }
            },
            testID: testID,
            value: value,
            style: {
              color: _$$_REQUIRE(_dependencyMap[6]).theme.colors.searchBarTextColor,
              fontFamily: _$$_REQUIRE(_dependencyMap[6]).theme.fonts.nunitoRegular,
              fontSize: 15,
              padding: 0
            }
          })
        })]
      }), autocomplete && autocomplete.length > 0 && /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[3]).jsx)(_$$_REQUIRE(_dependencyMap[4]).Box, {
        backgroundColor: "white",
        paddingY: "xxxs",
        children: autocomplete.map(function (item) {
          return /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[3]).jsx)(_$$_REQUIRE(_dependencyMap[7]).Button, {
            inline: true,
            onPress: function onPress() {
              if (onClickAutocomplete) {
                onClickAutocomplete(item);
              }
            },
            children: /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[3]).jsx)(_$$_REQUIRE(_dependencyMap[4]).Box, {
              width: "100%",
              paddingY: "nano",
              paddingLeft: "xxxs",
              children: /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[3]).jsx)(_$$_REQUIRE(_dependencyMap[8]).Typography, {
                variant: "botaoFiltrarEOrdenarProdutos",
                children: item
              })
            })
          }, `autocomplete-search-${item}`);
        })
      })]
    });
  }
