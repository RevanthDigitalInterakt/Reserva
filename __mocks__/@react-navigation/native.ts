const navigate = jest.fn();

const useNavigation = () => ({
  navigate,
});

export {
  useNavigation,
};
