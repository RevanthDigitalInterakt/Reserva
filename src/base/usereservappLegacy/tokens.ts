export const white = '#FFF';
export const fullBlack = '#000';

// cores primarias
export const preto = '#333333';
export const vermelhoFechadoRSV = '#5A0000';
export const marinho = '#13294B';
export const neutroFrio2 = '#8A8C8E';
export const neutroQuente2 = '#B1ACA8';
export const divider = '#D8D9DA';
export const dividerCounter = '#EAEAEA';
export const progressTextColor = '#979797';
export const searchBarTextColor = '#666666';

// cores secundarias
export const vermelhoRSV = '#A00000';
export const verde = '#22372B';
export const neutroQuente1 = '#D7CFCA';
export const offWhite = '#D6D1C4';
export const offWhite15 = '#F9F8F6';
export const whiteSecondary = '#FAFAFA';
export const neutroFrio1 = '#D8D9DA';
export const leve = '#F9E1DD';

// cores do sistema
export const vermelhoAlerta = '#EF1E1E';
export const verdeSucesso = '#38A238';
export const amareloAtencao = '#FEA528';
export const dropDownBorderColor = '#DDDDDD';

// larguras de borda
export const borderWidthNone = 0;
export const borderWidthHairline = 1;
export const borderWidthThin = 2;
export const borderWidthThick = 4;
export const borderWidthHeavy = 8;
export const borderDiscountTag = 12;

// cores de borda
export const borderButton = '#333333';
export const borderColorProduct = '#757575';

// backgrounds
export const backgroundMenuOpened = '#F1F1F1';
export const backgroundApp = '#F9F9F9';
export const modalBackDropColor = '#AFAFAF';
export const backgoundInput = '#F0F0F0';
export const backgoundDivider = '#F5F5F5';
