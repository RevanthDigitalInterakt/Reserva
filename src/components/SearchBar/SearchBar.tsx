import React from 'react';
import { TextInput, View } from 'react-native';
import { Box } from '../Box/Box';
import { IconLegacy } from '../IconLegacy/IconLegacy';
import { theme } from '../../base/usereservappLegacy/theme';
import { Button } from '../Button';
import { Typography } from '../Typography/Typography';
import { COLORS } from '../../base/styles';

interface SearchBarProps {
  height?: number;
  iconName?: string;
  placeholder?: string;
  value?: string;
  autocomplete?: string[];
  onClickIcon?: () => null;
  onClickAutocomplete?: (clickedItem: string) => null;
  onValueChange?: (value: string) => null;
  testID?: string;
}

export function SearchBar({
  height = 36,
  iconName,
  placeholder,
  autocomplete,
  value,
  onValueChange,
  onClickIcon,
  onClickAutocomplete,
  testID,
}: SearchBarProps) {
  return (
    <Box>
      <View
        style={{
          borderWidth: 1,
          borderColor: COLORS.LIGHT_GRAY,
          borderRadius: 10,
          height: 45,
          flexDirection: 'row',
          alignItems: 'center',
          justifyContent: 'space-between',
        }}
      >
        <TextInput
          placeholder="O que vocÃª procura?"
          onSubmitEditing={() => onClickIcon && onClickIcon()}
          style={{
            marginLeft: 10,
            fontSize: 14,
            padding: 0,
            fontFamily: theme.fonts.reservaSansRegular,
            fontWeight: 'normal',
            lineHeight: 14,
          }}
          onChangeText={(text) => {
            if (onValueChange) {
              onValueChange(text);
            }
          }}
        />
        <View style={{ marginRight: 20 }}>
          <IconLegacy name={iconName || 'Search'} size={18} />
        </View>
      </View>
      {autocomplete && autocomplete.length > 0 && (
        <Box backgroundColor="white" paddingY="xxxs">
          {autocomplete.map((item) => (
            <Button
              key={`autocomplete-search-${item}`}
              inline
              onPress={() => {
                if (onClickAutocomplete) {
                  onClickAutocomplete(item);
                }
              }}
            >
              <Box width="100%" paddingY="nano" paddingLeft="xxxs">
                <Typography variant="botaoFiltrarEOrdenarProdutos">
                  {item}
                </Typography>
              </Box>
            </Button>
          ))}
        </Box>
      )}
      {/* <Box
        height={height}
        flexDirection="row"
        justifyContent="space-between"
        alignItems="center"
        borderRadius="nano"
        backgroundColor="backgoundInput"
        paddingX="nano"
      >
        <Box
          ml="quarck"
          mr="nano"
        >
          <IconLegacy name={iconName || 'Search'} size={18} />
        </Box>
        <Box flexGrow={1}>
          <TextInput
            placeholderTextColor={theme.colors.searchBarTextColor}
            placeholder={placeholder}
            returnKeyType="search"
            enablesReturnKeyAutomatically
            onSubmitEditing={() => onClickIcon && onClickIcon()}
            onChangeText={(text) => {
              if (onValueChange) {
                onValueChange(text);
              }
            }}
            testID={testID}
            value={value}
            style={{
              color: theme.colors.searchBarTextColor,
              fontFamily: theme.fonts.nunitoRegular,
              fontSize: 15,
              padding: 0,
            }}
          />
        </Box>

      </Box>

      {autocomplete && autocomplete.length > 0 && (
      <Box backgroundColor="white" paddingY="xxxs">
        {autocomplete.map((item) => (
          <Button
            key={`autocomplete-search-${item}`}
            inline
            onPress={() => {
              if (onClickAutocomplete) {
                onClickAutocomplete(item);
              }
            }}
          >
            <Box width="100%" paddingY="nano" paddingLeft="xxxs">
              <Typography variant="botaoFiltrarEOrdenarProdutos">
                {item}
              </Typography>
            </Box>
          </Button>
        ))}
      </Box>
      )} */}
    </Box>
  );
}
