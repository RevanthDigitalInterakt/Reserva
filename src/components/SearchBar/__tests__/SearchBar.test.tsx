import React from 'react';
import { ThemeProvider } from 'styled-components/native';
import { MockedProvider } from '@apollo/client/testing';
import { render, fireEvent } from '@testing-library/react-native';
import { SearchBar } from '../SearchBar';
import { theme } from '../../../base/usereservappLegacy/theme';
import CartContextProvider from '../../../context/CartContext';

describe('SearchBar component', () => {
  const onValueChangeMock = jest.fn();

  const component = (
    <ThemeProvider theme={theme}>
      <MockedProvider addTypename={false}>
        <CartContextProvider>
          <SearchBar onValueChange={onValueChangeMock} />
        </CartContextProvider>
      </MockedProvider>
    </ThemeProvider>
  );

  it('renders correctly', () => {
    render(component);
  });

  it('displays placeholder correctly', () => {
    const placeholderText = 'Search';
    const { getByPlaceholderText } = render(<SearchBar placeholder={placeholderText} />);
    expect(getByPlaceholderText(placeholderText)).toBeDefined();
  });
  it('calls onValueChange correctly when text is changed', () => {
    const placeholderText = 'Search';
    const { getByPlaceholderText } = render(<SearchBar
      onValueChange={onValueChangeMock}
      placeholder={placeholderText}
    />);
    const textInput = getByPlaceholderText('Search');
    fireEvent.changeText(textInput, 'test');
    expect(onValueChangeMock).toHaveBeenCalledWith('test');
  });
  it('calls onClickAutocomplete correctly when an autocomplete item is clicked', () => {
    const onClickAutocompleteMock = jest.fn();
    const autocompleteItems = ['item1', 'item2', 'item3'];
    const { getByText } = render(
      <SearchBar autocomplete={autocompleteItems} onClickAutocomplete={onClickAutocompleteMock} />,
    );
    const firstAutocompleteItem = getByText('item1');
    fireEvent.press(firstAutocompleteItem);
    expect(onClickAutocompleteMock).toHaveBeenCalledWith('item1');
  });
});
