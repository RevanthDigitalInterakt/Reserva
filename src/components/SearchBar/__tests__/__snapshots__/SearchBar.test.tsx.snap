// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar component should match with the snapshot 1`] = `
<View
  style={{}}
>
  <View
    alignItems="center"
    backgroundColor="backgoundInput"
    borderRadius={5}
    flexDirection="row"
    height={36}
    justifyContent="space-between"
    paddingX="nano"
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#F0F0F0",
        "borderBottomLeftRadius": 5,
        "borderBottomRightRadius": 5,
        "borderTopLeftRadius": 5,
        "borderTopRightRadius": 5,
        "flexDirection": "row",
        "height": 36,
        "justifyContent": "space-between",
        "paddingLeft": 8,
        "paddingRight": 8,
      }
    }
  >
    <View
      ml="quarck"
      mr="nano"
      style={
        {
          "marginLeft": 4,
          "marginRight": 8,
        }
      }
    >
      <RNSVGSvgView
        bbHeight="18"
        bbWidth="18"
        color="preto"
        focusable={false}
        size={18}
        style={
          [
            {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            {
              "color": "#333333",
              "height": 18,
              "width": 18,
            },
            {
              "flex": 0,
              "height": 18,
              "width": 18,
            },
          ]
        }
        tintColor="preto"
      >
        <RNSVGGroup
          fill={
            {
              "payload": 4278190080,
              "type": 0,
            }
          }
        >
          <RNSVGSvgView
            align="xMidYMid"
            bbHeight="100%"
            bbWidth="100%"
            focusable={false}
            height="100%"
            meetOrSlice={0}
            minX={0}
            minY={0}
            style={
              [
                {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                {
                  "height": "100%",
                  "width": "100%",
                },
                {
                  "flex": 0,
                  "height": "100%",
                  "width": "100%",
                },
              ]
            }
            vbHeight={17.112}
            vbWidth={17.493}
            width="100%"
            xmlns="http://www.w3.org/2000/svg"
          >
            <RNSVGGroup
              fill={
                {
                  "payload": 4278190080,
                  "type": 0,
                }
              }
            >
              <RNSVGGroup
                fill={null}
                matrix={
                  [
                    1,
                    0,
                    0,
                    1,
                    -283.354,
                    -17.548,
                  ]
                }
                propList={
                  [
                    "fill",
                    "stroke",
                    "strokeWidth",
                    "strokeLinecap",
                    "strokeLinejoin",
                  ]
                }
                stroke={
                  {
                    "payload": 4281545523,
                    "type": 0,
                  }
                }
                strokeLinecap={1}
                strokeLinejoin={1}
                strokeWidth="1"
              >
                <RNSVGEllipse
                  cx="7.227"
                  cy="7.227"
                  fill={
                    {
                      "payload": 4278190080,
                      "type": 0,
                    }
                  }
                  matrix={
                    [
                      1,
                      0,
                      0,
                      1,
                      283.854,
                      18.048,
                    ]
                  }
                  rx="7.227"
                  ry="7.227"
                />
                <RNSVGPath
                  d="M0 0L3.762 3.762"
                  fill={
                    {
                      "payload": 4278190080,
                      "type": 0,
                    }
                  }
                  matrix={
                    [
                      1,
                      0,
                      0,
                      1,
                      296.377,
                      30.19,
                    ]
                  }
                />
              </RNSVGGroup>
            </RNSVGGroup>
          </RNSVGSvgView>
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
    <View
      flexGrow={1}
      style={
        {
          "flexGrow": 1,
        }
      }
    >
      <TextInput
        onChangeText={[Function]}
        onSubmitEditing={[Function]}
        placeholderTextColor="#666666"
        returnKeyType="search"
        style={
          {
            "color": "#666666",
            "fontFamily": "Nunito-Regular",
            "fontSize": 15,
            "padding": 0,
          }
        }
      />
    </View>
  </View>
</View>
`;
